django_socio_grpc/servicer_proxy.py

```diff
@@ -1,130 +1,252 @@
+    adapt_method_mode = BaseHandler.adapt_method_mode
+
+    def load_middleware(self, is_async=False):
+        """
+        Populate middleware lists from settings.GRPC_MIDDLEWARE.
+        Must be called after the environment is fixed (see __call__ in subclasses).
+        """
+
+        handler = self._get_response_async if is_async else self._get_response
+        handler_is_async = is_async
+        for middleware_path in reversed(grpc_settings.GRPC_MIDDLEWARE):
+            middleware = import_string(middleware_path)
+            middleware_can_sync = getattr(middleware, "sync_capable", True)
+            middleware_can_async = getattr(middleware, "async_capable", False)
+            if not middleware_can_sync and not middleware_can_async:
+                raise RuntimeError(
+                    "Middleware %s must have at least one of "
+                    "sync_capable/async_capable set to True." % middleware_path
+                )
+            elif not handler_is_async and middleware_can_sync:
+                middleware_is_async = False
+            else:
+                middleware_is_async = middleware_can_async
+            try:
+                # Adapt handler, if needed.
+                adapted_handler = self.adapt_method_mode(
+                    middleware_is_async,
+                    handler,
+                    handler_is_async,
+                    debug=settings.DEBUG,
+                    name="middleware %s" % middleware_path,
+                )
+                mw_instance = middleware(adapted_handler)
+            except MiddlewareNotUsed as exc:
+                if settings.DEBUG:
+                    if str(exc):
+                        logger.debug("MiddlewareNotUsed(%r): %s", middleware_path, exc)
+                    else:
+                        logger.debug("MiddlewareNotUsed: %r", middleware_path)
+                continue
+            else:
+                handler = adapted_handler
+
+            if mw_instance is None:
+                raise ImproperlyConfigured(
+                    "Middleware factory %s returned None." % middleware_path
+                )
+
+            handler = mw_instance
+            handler_is_async = middleware_is_async
+
+        # Adapt the top of the stack, if needed.
+        handler = self.adapt_method_mode(is_async, handler, handler_is_async)
+        # We only assign to this when initialization is complete as it is used
+        # as a flag for initialization being complete.
+        self._middleware_chain = handler
+
+    @abc.abstractmethod
+    def _get_response(self, request: GRPCRequestContainer):
+        ...
+
+    @abc.abstractmethod
+    async def _get_response_async(self, request: GRPCRequestContainer):
+        ...
+
+    @abc.abstractmethod
+    def process_exception(self, exception, request):
+        ...
+
+    @abc.abstractmethod
+    async def async_process_exception(self, exception, request):
+        ...
+
+
+class ServicerProxy(MiddlewareCapable):
```

Comment:
can you add a comment here describing the flow process like:
__getattr__ -------> get_handler ---async true & stream true------> _get_async_stream_handler
                                                   ---async true & stream false ----> _get_async_handler
                                                   --- async false ----------------------> _get_handler

etc it will be much easier to understand code and make change to external developper